### PR TITLE
Fix parsing of GVFS-aware Git version numbers that contain - (for PR validation builds)

### LINKED ISSUE #175: GVFS-aware Git versions containing '-' don't get parsed correctly
Our PR builds for GVFS-aware Git end with '-pr' and that version number isn't correctly parsed by our parsing scripts. We should follow the spec that NuGet defines for what defines a package number: https://docs.microsoft.com/en-us/nuget/reference/package-versioning

### PR DESCRIPTION
**Pull request by nickgra**
Correct regex used to match GVFS-aware Git package to actually match the NuGet spec for what defines a package. See https://docs.microsoft.com/en-us/nuget/reference/package-versioning

This fixes #175

### CODE REVIEW

**Comment by sanoursa** [comment]
The fact that this same regex shows up in 3 separate scripts could end up being a maintenance issue. Can it be isolated into just one script that returns the appropriate version number?

File: Scripts/Mac/BuildGVFSForMac.sh [hunk 1]
```diff
@@ -34,7 +34,7 @@ fi
 
 $SCRIPTDIR/DownloadGVFSGit.sh || exit 1
 GVFSPROPS=$SRCDIR/GVFS/GVFS.Build/GVFS.props
-GITVERSION="$(cat $GVFSPROPS | grep GitPackageVersion | grep -Eo '[0-9.]{1,}')"
+GITVERSION="$(cat $GVFSPROPS | grep GitPackageVersion | grep -Eo '[0-9.]+(-\w+)?')"
```
Comment (wilbaker) [inline Scripts/Mac/BuildGVFSForMac.sh]: I'm not familiar with bash regex syntax, what does `+(-\w+)?` match against?

**Comment by nickgra** [comment]
@sanoursa I agree that it's no good to have this code duplicated in three places. In the past we talked about writing out the version number once to the build directory and having other scripts just read it from that file. Our intention was to make sure that we weren't using a stale version number (imagine the version has changed since your last build, and you're running the Prep script before ever running the Build script that would create the file), but I'm not sure if that scenario is going to ever be common.

I can make that change here.

**Comment by sanoursa** [comment]
@nickgra isn't it possible to call a script that returns a value?

File: Scripts/Mac/BuildGVFSForMac.sh (hunk 1 above)
Comment (wilbaker) [reply Scripts/Mac/BuildGVFSForMac.sh]: What does the `\w` match? (also where did you find the spec?)

**Comment by nickgra** [comment]
@sanoursa Yup, that approach works too. I took that approach for commit 2.

File: Scripts/Mac/BuildGVFSForMac.sh (hunk 1 above)
Comment (nickgra) [reply Scripts/Mac/BuildGVFSForMac.sh]: The \w matches any word (alphanumeric) character, so we're matching any number of words after the -. So -pr, -rc, -release1 would all be valid values, but something like -release-1 wouldn't be.

The 'spec' of sorts is here: https://docs.microsoft.com/en-us/nuget/reference/package-versioning

**Pull request merged**